front/src/modules/ui/input/relation-picker/components/SingleEntitySelectBase.tsx

```diff
@@ -38,26 +41,39 @@ export const SingleEntitySelectBase = <
   onCancel,
   onEntitySelected,
   selectedEntity,
+  onCreateHandler,
+  showCreateButton,
 }: SingleEntitySelectBaseProps<CustomEntityForSelect>) => {
   const containerRef = useRef<HTMLDivElement>(null);
   const entitiesInDropdown = [selectedEntity, ...entitiesToSelect].filter(
     (entity): entity is CustomEntityForSelect =>
       assertNotNull(entity) && isNonEmptyString(entity.name.trim()),
   );
 
-  const { hoveredIndex, resetScroll } = useEntitySelectScroll({
-    entities: entitiesInDropdown,
+  const { hoveredId, resetScroll } = useEntitySelectScroll({
+    hoverIds: [
+      ...entitiesInDropdown.map((item) => item.id),
+      ...(onCreateHandler && showCreateButton ? [CreateButtonId] : []),
+    ],
     containerRef,
   });
 
   useScopedHotkeys(
     Key.Enter,
     () => {
-      onEntitySelected(entitiesInDropdown[hoveredIndex]);
+      if (onCreateHandler && hoveredId === CreateButtonId) {
+        onCreateHandler();
+      } else {
+        const index = entitiesInDropdown.findIndex(
+          (val) => val.id === hoveredId,
```

Comment:
Always try to keep a clear naming even for small parts of the code, here it should be at least `entity`